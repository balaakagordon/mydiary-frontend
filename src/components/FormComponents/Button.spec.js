import React from 'react';
import { mount } from 'enzyme';
import Button from './Button';

describe('The Button component', () => {
    let component

    beforeAll(() => {
        component = mount(<Button />)
    })

    it('should render without crashing', () => {
        expect(component).toBeDefined();
    })

    it('should match the snapshot', () => {
        expect(component).toMatchSnapshot();
    })
})

describe('Button component functions', () => {
    const buttonAction = jest.fn();

    it('should call action when clicked', () => {
        let wrapper = mount(
            <Button
                action={buttonAction}
                title={'Submit'}
                className={'class'}
            />
        );
        wrapper.find('button').simulate('click');
        expect(buttonAction).toHaveBeenCalled();
    })
})
