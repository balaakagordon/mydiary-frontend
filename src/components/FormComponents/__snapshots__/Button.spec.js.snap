// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Button component should match the snapshot 1`] = `
<Button>
  <button />
</Button>
`;
